Support ethermint mode with features

thomas-nguy opened the pull request:
- update deep_space with ethermint feature
- add a PrivateKey which switch to ethermint when ethermint feature is turn on
- change default hd_path if ethermint feature turn on


To build with ethermint feature run :
` cargo build --features ethermint `

yihuang reviewed (approved):
it's the same patch as this one: https://github.com/crypto-org-chain/gravity-bridge/pull/31, which is verified in cronos's integration tests.

EricBolten commented on orchestrator/Cargo.toml:
```diff
@@ -41,3 +41,15 @@ gorc = { path = "./gorc" }
+]
+
+[patch.crates-io]
+deep_space = { git = "https://github.com/iqlusioninc/deep_space/", branch = "master" }
```
Unless I'm understanding this incorrectly, isn't this patch directive going to override all the deep_space dependency changes in the individual `Cargo.toml` files?

yihuang replied:
Because `patch.crates-io` only works when the crate `Cargo.toml` depends on a regular dependency and the patched one has the same version number as it.

EricBolten replied:
Right, I guess what I don't understand is how this is different from leaving the original dependency in place rather than changing it and then patching back to the original version.

yihuang replied:
> Right, I guess what I don't understand is how this is different from leaving the original dependency in place rather than changing it and then patching back to the original version.

I see, that's because it wasn't `master` branch before our patch get merged into master ;D , and in this way, we only need to change one place to switch the repo and branch, so I think it's good to keep it this way.

thomas-nguy replied:
I guess it does make it cleaner that way.

Is it fine to keep this refactoring @EricBolten or do you prefer to revert the change?

zmanian replied:
I wasn't familiar with this Cargo feature. I think it's fine. It does make things simpler.

EricBolten reviewed (commented):
Question regarding patching.

thomas-nguy commented:
@jackzampolin @EricBolten I have reverted the previous cargo.lock change and just updated with minimal changes. It should be more manageable to review 

thanks @yihuang for the effort
